tests/Twig/Extension/SonataAdminExtensionTest.php

```diff
@@ -2290,6 +2292,199 @@ public function testGetXEditableChoicesIsIdempotent(array $input)
         );
     }
 
+    public function select2LocalesProvider()
+    {
+        return [
+            ['ar', 'ar'],
+            ['az', 'az'],
+            ['bg', 'bg'],
+            ['ca', 'ca'],
+            ['cs', 'cs'],
+            ['da', 'da'],
+            ['de', 'de'],
+            ['el', 'el'],
+            ['en', null],
+            ['es', 'es'],
+            ['et', 'et'],
+            ['eu', 'eu'],
+            ['fa', 'fa'],
+            ['fi', 'fi'],
+            ['fr', 'fr'],
+            ['gl', 'gl'],
+            ['he', 'he'],
+            ['hr', 'hr'],
+            ['hu', 'hu'],
+            ['id', 'id'],
+            ['is', 'is'],
+            ['it', 'it'],
+            ['ja', 'ja'],
+            ['ka', 'ka'],
+            ['ko', 'ko'],
+            ['lt', 'lt'],
+            ['lv', 'lv'],
+            ['mk', 'mk'],
+            ['ms', 'ms'],
+            ['nb', 'nb'],
+            ['nl', 'nl'],
+            ['pl', 'pl'],
+            ['pt', 'pt-PT'],
+            ['pt-BR', 'pt-BR'],
+            ['pt-PT', 'pt-PT'],
+            ['ro', 'ro'],
+            ['rs', 'rs'],
+            ['ru', 'ru'],
+            ['sk', 'sk'],
+            ['sv', 'sv'],
+            ['th', 'th'],
+            ['tr', 'tr'],
+            ['ug', 'ug-CN'],
+            ['ug-CN', 'ug-CN'],
+            ['uk', 'uk'],
+            ['vi', 'vi'],
+            ['zh', 'zh-CN'],
+            ['zh-CN', 'zh-CN'],
+            ['zh-TW', 'zh-TW'],
+        ];
+    }
+
+    /**
+     * @dataProvider select2LocalesProvider
+     */
+    public function testCanonicalizedLocaleForSelect2($original, $expected)
```

Comment:
please use $expected as the first argument, like in the assertations.

Same for the other method